core/src/components/picker-column/picker-column.tsx

```diff
@@ -87,10 +88,35 @@ export class PickerColumnCmp implements ComponentInterface {
       // We perfom a DOM read over a rendered item, this needs to happen after the first render
       this.optHeight = colEl.firstElementChild ? colEl.firstElementChild.clientHeight : 0;
     }
-
     this.refresh();
   }
 
+  componentShouldUpdate(newVal: any, oldVal: any, propName: string): boolean | void {
+    // Check if the options have changed so `refresh()` isn't called on
+    // each column change (ex: `setSelected()` method)
+    // The change is most likely triggered by an outside source
+    // For example, when adding or removing an option dynamically, etc.
+    // If the dynamic option was added then it needs to be added to the DOM
+    // without animation
+    // Otherwise the animation will be render the option
+    // at the top of the column and then it will fall down to its place
+    if (propName === 'col' && JSON.stringify(newVal.options) !== JSON.stringify(oldVal.options)) {
+      this.shouldAnimteOnRender = false;
+    }
+  }
```

Comment:
Can we set `this.shouldAnimateOnRender` in the `@Watch` callback? That callback runs before the component re-renders and will only fire when the `col` data changes.

Reply:
The `col` data changes can also be triggered when a user toggles through the picker. Is there a way to compare `oldVal` and `newVal` in `@Watch`? I didn't see an example on the Stencil docs.

Reply:
I had my function order wrong. You're right that we don't need to compare. It'll be changed in the next commit.